nemo/collections/asr/data/audio_to_label.py

```diff
@@ -825,3 +1005,765 @@ def sliced_seq_collate_fn(self, batch):
+                Representative speaker label for each segment. This variable only has one speaker label for each base-scale segment.
+
+        """
+        seg_target_list, base_clus_label = [], []
+        self.scale_n = len(self.ms_ts_dict[uniq_id]['scale_dict'])
+        subseg_time_stamp_list = self.ms_ts_dict[uniq_id]["scale_dict"][self.scale_n - 1]["time_stamps"]
+        for line in subseg_time_stamp_list:
+            line_split = line.split()
+            seg_stt, seg_end = float(line_split[0]), float(line_split[1])
+            seg_stt_fr, seg_end_fr = int(seg_stt * self.fr_per_sec), int(seg_end * self.fr_per_sec)
+            soft_label_vec = torch.sum(fr_level_target[seg_stt_fr:seg_end_fr, :], axis=0) / (seg_end_fr - seg_stt_fr)
+            label_int = torch.argmax(soft_label_vec)
+            label_vec = (soft_label_vec > self.soft_label_thres).float()
+            seg_target_list.append(label_vec.detach())
+            base_clus_label.append(label_int.detach())
+        seg_target = torch.stack(seg_target_list)
+        base_clus_label = torch.stack(base_clus_label)
+        return seg_target, base_clus_label
+
+    def parse_rttm_for_ms_targets(self, sample, tup_spks=None):
+        """
+        Generate target tensor variable by extracting groundtruth diarization labels from an RTTM file.
+        This function converts (start, end, speaker_id) format into base-scale (the finest scale) segment level
+        diarization label in a matrix form.
+
+        Example of seg_target:
+            [[0., 1.], [0., 1.], [1., 1.], [1., 0.], [1., 0.], ..., [0., 1.]]
+
+        Args:
+            sample:
+                DiarizationSpeechLabel instance containing the following variables.
+
+                audio_file (str):
+                    Path of the input audio file (raw waveform).
+                offset (str):
+                    Offset of the input audio file provided in the input manifest file.
+                duration (float):
+                    Duration of the input audio file provided in the input manifest file.
+                rttm_file (str):
+                    Path of the groundtruth diarization annotation file (RTTM format).
+
+        Returns:
+            clus_label_index (torch.tensor):
+                Groundtruth Clustering label (cluster index for each segment) from RTTM files for training purpose.
+            seg_target  (torch.tensor):
+                Tensor varialble containing hard-labels of speaker activity in each base-scale segment.
+            scale_mapping (torch.tensor):
+                Matrix containing the segment indices of each scale. scale_mapping is necessary for reshaping the
+                multiscale embeddings to form an input matrix for the MSDD model.
+
+        """
+        rttm_lines = open(sample.rttm_file).readlines()
+        uniq_id = self.get_uniq_id_with_range(sample)
+        rttm_timestamps = self.extract_seg_info_from_rttm(uniq_id, rttm_lines, tup_spks=None)
```

Comment:
if tup_spks is not used in the function, you should remove the input argument

Reply:
Removed it.